File: String/add two large integers, which cannot be represented by built in data types.c
#include<stdio.h>
#include<string.h>
int main(){
    char num1[100],num2[100];
    int len1,len2,a[100],b[100],i=0,sum[101],carry=0,rs=0;
    scanf("%s",&num1);
    scanf("%s",&num2);
    len1=strlen(num1);
    len2=strlen(num2);
    while(num1[i]!='\0'){
        a[i]=num1[i]-48;
        i++;
    }
    i=0;
    while(num2[i]!='\0'){
        b[i]=num2[i]-48;
        i++;
    }

    if(len1>len2){

        len1=len1-1;
        len2=len2-1;

        for(i=len2;i>=0;i--){
            sum[rs]=(a[len1]+b[i]+carry)%10;
            carry=(a[len1]+b[i]+carry)/10;
            len1--;
            rs++;
        }

        while(len1>=0){
            sum[rs]=(a[len1]+carry)%10;
            carry=(a[len1]+carry)/10;
            len1--;
            rs++;
        }
        rs=rs-1;

        if(carry!=0){
            sum[rs]=carry;
        }

    }
    else if(len2>len1){

        len1=len1-1;
        len2=len2-1;

        for(i=len1;i>=0;i--){
            sum[rs]=(b[len2]+a[i]+carry)%10;
            carry=(b[len2]+a[i]+carry)/10;
            len2--;
            rs++;
        }

        while(len2>=0){
            sum[rs]=(b[len2]+carry)%10;
            carry=(b[len2]+carry)/10;
            len2--;
            rs++;
        }
        rs=rs-1;

        if(carry!=0){
            rs++;
            sum[rs]=carry;
        }

    }

    else{
        len1=len1-1;
        len2=len2-1;

        for(i=len2;i>=0;i--){
            sum[rs]=(a[len1]+b[i]+carry)%10;
            carry=(a[len1]+b[i]+carry)/10;
            rs++;
            len1--;
        }
        rs=rs-1;

        if(carry!=0){
            rs++;
            sum[rs]=carry;
        }


    }
    for(i=rs;i>=0;i--){
        printf("%d",sum[i]);
    }

}

File: String/add  two large integers, which cannot be represented by built in data types.py
a=input("Number 1 = ")
b=input("Number 2 = ")
a=list(a)
b=list(b)
carry=0
le1=len(a)
le2=len(b)
sum=[]
if le1>le2:
    while le2>0:
        le2=le2-1
        le1=le1-1
        sum.append(int((int(a[le1])+int(b[le2])+carry)%10))
        carry=int((int(a[le1])+int(b[le2])+carry)/10)
    while le1>0:
        le1=le1-1
        sum.append(int((int(a[le1])+carry)%10))
        carry=int((int(a[le1])+carry)/10)
    if carry<0:
        sum.append(int(carry))
if le1<le2:
    while le1>0:
        le2=le2-1
        le1=le1-1
        sum.append(int((int(b[le2])+int(a[le1])+carry)%10))
        carry=int((int(b[le2])+carry)/10)
    while le2>0:
        le2=le2-1
        sum.append(int(int(b[le2])+carry)%10)
        carry=int((int(b[le2])+carry)/10)
    if carry<0:
        sum.append(carry)
if le1==le2:
    while le2>0:
        le2=le2-1
        le1=le1-1
        sum.append(int((int(a[le1])+int(b[le2])+carry)%10))
        carry=int((int(a[le1])+int(b[le2])+carry)/10)
    if carry<0:
        sum.append(int(carry))
length=len(sum)      
while length!=0:
    length=length-1
    print(sum[length],end="")
    
        

            
